src/Toolkit/Toolkit/UI/Controls/PopupViewer/PopupViewer.Windows.cs

```diff
@@ -56,6 +56,32 @@ private static void OnPopupManagerPropertyChanged(DependencyObject d, Dependency
             ((PopupViewer)d).Popup = ((PopupViewer)d).PopupManager?.Popup;
 #pragma warning restore CS0618 // Type or member is obsolete
         }
+
+        internal static PopupViewer? GetPopupViewerParent(DependencyObject? child) => GetParent<PopupViewer>(child);
+
+        private static T? GetParent<T>(DependencyObject? child) where T : DependencyObject
+        {
+            // Traverse up the visual tree to find the PopupViewer parent
+            while (child is not null)
+            {
+                if (child is FrameworkContentElement contentElement)
+                {
+                    // Use the logical parent for FrameworkContentElement instances
+                    child = contentElement.Parent ?? VisualTreeHelper.GetParent(contentElement);
+                }
```

Comment:
this rewrite is quite confusing to me - there's now two `VisualTreeHelper.GetParent` calls in this method. I think it was clearer before how we had it. The fallback would never work because using the VisualTreeHelper.GetParent on a FrameworkContentElement was clearly throwing when we first tested it.
Perhaps all you really need to do is remove the fallback.

Reply:
```
internal static PopupViewer? GetPopupViewerParent(DependencyObject? child)
{
    if (child is FrameworkContentElement elm)
    {
        child = GetVisualParent(elm);
    }
    if (child is null)
        return null;
    var parent = VisualTreeHelper.GetParent(child);
    while (parent is not null and not PopupViewer)
    {
        parent = VisualTreeHelper.GetParent(parent);
    }
    return parent as PopupViewer;
}
internal static Visual? GetVisualParent(FrameworkContentElement child)
{
    var parent = child.Parent;
    while (parent is FrameworkContentElement elm)
    {
        parent = elm.Parent;
    }
    return parent as Visual;
}

```

This was the original implementation, Its just refactor and had two `VisualTreeHelper.GetParent` calls.

Reply:
It's not the same. If `VisualTreeHelper.GetParent(contentElement)` is ever hit it'll throw an exception.

Reply:
Got it. Reverting it back.